code/model/EditableCustomRule.php

```diff
@@ -147,4 +151,91 @@ public function canUnpublish($member = null)
     {
         return $this->canDelete($member);
     }
-}
+
+    /**
+     * Substitutes configured rule logic with it's JS equivalents and returns them as array elements
+     * @return array
+     */
+    public function buildExpression()
+    {
+        /** @var EditableFormField $formFieldWatch */
+        $formFieldWatch = $this->ConditionField();
+        //Encapsulated the action to the object
+        $action = $formFieldWatch->getJsEventHandler();
+
+        // is this field a special option field
+        $checkboxField = $formFieldWatch->isCheckBoxField();
+        $radioField = $formFieldWatch->isRadioField();
+        $target = sprintf('$("%s")', $formFieldWatch->getSelectorFieldOnly());
+        $fieldValue = Convert::raw2js($this->FieldValue);
+
+        $conditionOptions = array(
+            'ValueLessThan'         => '<',
+            'ValueLessThanEqual'    => '<=',
+            'ValueGreaterThan'      => '>',
+            'ValueGreaterThanEqual' => '>='
+        );
+        // and what should we evaluate
+        switch ($this->ConditionOption) {
+            case 'IsNotBlank':
+            case 'IsBlank':
+                $expression = ($checkboxField || $radioField) ? "{$target}.is(\":checked\")" : "{$target}.val() == ''";
```

Comment:
Incorrect logic here for checkbox fields; You are treating `:checked` as IsBlank; The inverse is true.